Automattic/wp-calypso
# Jetpack Manage: Pricing page -adding a more about link leading to a LicenseLightbox

Related to https://github.com/Automattic/jetpack-manage/issues/258

## Proposed Changes

* This PR adds a 'More about' link to all product cards on the Manage Pricing page. 
* When clicked, this opens a LicenseLightbox for each product, with the correct price showing.
* Clicking on the 'Select License' link should lead to the issue license page, opening the 'Review License' lightbox for the relevant product.

Additional note: I spent some time looking into how to prevent the duplication between files but did not come up with anything helpful up until this point. I'm opening this up for review with that in mind, and open to suggestions.

## Testing Instructions

* To test, check out this branch. 
* Visit `http://jetpack.cloud.localhost:3000/manage/pricing`, and click on the 'More about' link for any product. Try for bundles as well (you'll need to manually add a bundleSize in `/manage/pricing/license-products-list/all-license-items.tsx` and `/manage/pricing/license-products-list/most-popular-plans.tsx` in that case).
* In all cases, the lightbox should open and show the correct amount.
* Clicking on 'Select License' should take you to the issue license page, opening the 'Review License' lightbox for the relevant product (barring issues related to initial load being investigated separately - you may need to refresh the page to confirm the lightbox will show on the issue licenses page).

<img width="1047" alt="Screenshot 2024-01-31 at 20 37 29" src="https://github.com/Automattic/wp-calypso/assets/16754605/4d1bf5d8-2526-4bb7-a3df-263530b4288a">


## Pre-merge Checklist
